### DISCUSSION TITLE
Problem with reading HLS source from Netatmo Outdoor camera

### DISCUSSION THREAD

**Discussion by tlowcode** [bluenviron/mediamtx#1366] [Questions]
I'm struggling to correctly open a HLS stream of a Netatmo Outdoor camera:

I always see error like that:

2023/01/07 22:25:05 DEB [path proxied] [hls source] downloading primary playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
2023/01/07 22:25:05 DEB [path proxied] [hls source] primary playlist is a stream playlist
2023/01/07 22:25:05 DEB [path proxied] [hls source] downloading stream playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
2023/01/07 22:25:05 DEB [path proxied] [hls source] downloading segment http://192.168.20.14/TOKEN/live/files/low/live0000296815.ts
2023/01/07 22:25:06 DEB [path proxied] [hls source] downloading stream playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
2023/01/07 22:25:06 INF [path proxied] [hls source] ready: 2 tracks (H264, MPEG4-audio)
2023/01/07 22:25:06 INF [HLS] [muxer proxied] created automatically
2023/01/07 22:25:06 INF [HLS] [muxer proxied] is converting into HLS, 2 tracks (H264, MPEG4-audio)
2023/01/07 22:25:06 DEB [path proxied] [hls source] segment inverse position: 2
2023/01/07 22:25:06 DEB [path proxied] [hls source] downloading segment http://192.168.20.14/TOKEN/live/files/low/live0000296816.ts
2023/01/07 22:25:07 DEB [path proxied] [hls source] downloading stream playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
2023/01/07 22:25:07 DEB [path proxied] [hls source] segment inverse position: 2
2023/01/07 22:25:07 DEB [path proxied] [hls source] downloading segment http://192.168.20.14/TOKEN/live/files/low/live0000296817.ts
2023/01/07 22:25:08 DEB [path proxied] [hls source] downloading stream playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
2023/01/07 22:25:08 DEB [path proxied] [hls source] segment inverse position: 1
2023/01/07 22:25:08 DEB [path proxied] [hls source] downloading segment http://192.168.20.14/TOKEN/live/files/low/live0000296818.ts
2023/01/07 22:25:09 DEB [path proxied] [hls source] downloading stream playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
2023/01/07 22:25:10 DEB [path proxied] [hls source] segment inverse position: 1
2023/01/07 22:25:10 DEB [path proxied] [hls source] downloading segment http://192.168.20.14/TOKEN/live/files/low/live0000296819.ts
2023/01/07 22:25:10 DEB [path proxied] [hls source] downloading stream playlist http://192.168.20.14/TOKEN/live/files/low/index.m3u8
> 2023/01/07 22:25:10 INF [path proxied] [hls source] ERR: following segment not found or not ready yet
I've never managed to establish a working stream (always fails on 1st or 2nd read of source)

Stream playlist looks like this:

#EXTM3U
#EXT-X-VERSION:7
#EXT-X-MEDIA-SEQUENCE:296939
#EXT-X-TARGETDURATION:2
#EXTINF:2.000000,
live0000296939.ts
#EXTINF:2.000000,
live0000296940.ts
#EXTINF:2.000000,
live0000296941.ts

I don't know if this issue is related to the camera itself but could be solved e.g. using a retry mechanism or introduction of a delay?

The issue occurred setting some attributes of HLS config (e.g. segments 3, duration 2s, ..)
When I'm only using path config the applications crashes:

`2023/01/07 22:43:08 INF [path netatmo1] [hls source] ready: 2 tracks (H264, MPEG4-audio)
panic: runtime error: index out of range [-1]

goroutine 25 [running]:
github.com/aler9/rtsp-simple-server/internal/hls.findSegmentWithID(...)
/s/internal/hls/client_downloader_stream.go:43
github.com/aler9/rtsp-simple-server/internal/hls.(*clientDownloaderStream).fillSegmentQueue(0xc000126070, {0xed48c0, 0xc000404100}, 0x1?)
/s/internal/hls/client_downloader_stream.go:230 +0x36d
github.com/aler9/rtsp-simple-server/internal/hls.(*clientDownloaderStream).run(0xc000126070, {0xed48c0, 0xc000404100})
/s/internal/hls/client_downloader_stream.go:148 +0x45b
github.com/aler9/rtsp-simple-server/internal/hls.(*clientRoutinePool).add.func1()
/s/internal/hls/client_routine_pool.go:44 +0x87
created by github.com/aler9/rtsp-simple-server/internal/hls.(*clientRoutinePool).add
/s/internal/hls/client_routine_pool.go:41 +0xaa
`

Thank you in advance!

**Comment by aler9** [comment]
Hello, the crash has been fixed in 31541765fd837493828f782d12cf84f985d0d40b and v0.21.1, please use Issues in order to report issues with cameras.

[a comment by an automated account is not shown]
